src/main/java/com/powsybl/openloadflow/network/impl/LfStaticVarCompensatorImpl.java

```diff
@@ -102,9 +107,10 @@ public double getMaxP() {
 
     @Override
     public void updateState() {
+        double vSquare = bus.getV() * bus.getV() * bus.getNominalV() * bus.getNominalV();
         svc.getTerminal()
                 .setP(0)
-                .setQ(Double.isNaN(calculatedQ) ? svc.getReactivePowerSetpoint() : -calculatedQ);
+                .setQ((Double.isNaN(calculatedQ) ? -targetQ : -calculatedQ) - optionalB * vSquare);
```

Comment:
Here: in case of calculatedQ equals to NaN, it means that the static var compensator does not regulate voltage and so, the optional B should be equal to zero.